### DISCUSSION TITLE
SSD365VSAddIn for VS2022

### DISCUSSION THREAD

**Discussion by ReneVolkmer** [shashisadasivan/SSD365VSAddIn#65] [General]
Hello Sashi,
do you plan to upgrade this great AddIn to work with VS2022 please?

Best regards
Rene

**Comment by shashisadasivan** [comment]
@ReneVolkmer yes i would love to, but currently a bit restricted on resources

**Reply by ReneVolkmer** [reply]
@shashisadasivan  I updated references to correspond with VS2022, but I am getting "cannot convert from 'EnvDTE.DTE' to 'System.IServiceProvider'	SSD365VSAddIn" from https://github.com/shashisadasivan/SSD365VSAddIn/blob/85a53c1d6de6c4268836c4c8e72eab21d7456c0a/SSD365VSAddIn/SSD365VSAddIn/Common/CommonUtil.cs#L49 . Using DTE2 (as described here https://learn.microsoft.com/en-us/visualstudio/extensibility/migration/breaking-api-list?view=vs-2022#dte-and-_dte-type-cast-failures) does not help.  VSApplicationContext implementation seems not to be changed. Any hint please?

<img width="938" alt="image" src="https://github.com/shashisadasivan/SSD365VSAddIn/assets/17122845/d8f695c3-87df-4f74-a8dd-8b218854cf6c">

**Reply by ReneVolkmer** [reply]
Explicit casting results in to runtime error "Unable to cast COM object of type 'System.__ComObject' to interface type 'System.IServiceProvider'. This operation failed because the QueryInterface call on the COM component for the interface with IID '{8F10F540-7F5D-3F37-8D79-1E0AEB074AA0}' failed due to the following error: No such interface supported (Exception from HRESULT: 0x80004002 (E_NOINTERFACE))."
<img width="714" alt="image" src="https://github.com/shashisadasivan/SSD365VSAddIn/assets/17122845/22381269-b9f7-47e1-9745-83930d37c7be">

**Reply by shashisadasivan** [reply]
I have managed to get this working. Will be putting this change in to a different branch

**Reply by ReneVolkmer** [reply]
hello @shashisadasivan, great news, thank you!

**Reply by shashisadasivan** [reply]
I have created issue #66 for this.
This should now compile and show up in the extensions in VS2022

**Reply by ReneVolkmer** [reply]
label management functionality returns error:
![image](https://github.com/shashisadasivan/SSD365VSAddIn/assets/17122845/8c036b7f-2d45-4d3b-9136-6b16657fe13a)

**Reply by shashisadasivan** [reply]
@ReneVolkmer Please take the latest code and try again. This is working on my application based on the last test.
I will also close this discussion. PLease post any issues on Issues #66

**Comment by shashisadasivan** [comment]
@ReneVolkmer how have you been tracking along with this ?

**Reply by ReneVolkmer** [reply]
Hello @shashisadasivan,
I deployed fresh devbox on 10.0.40 with VS2022 (17.10.2), when using https://raw.githubusercontent.com/shashisadasivan/SSD365VSAddIn/master/Misc/install.ps1 for VS Addin installation (only Newtonsoft.Json.dll and SSD365VSAddIn.dll, with "Unblock" property checked), I am getting "No addins discovered:
![image](https://github.com/shashisadasivan/SSD365VSAddIn/assets/17122845/6a5e249a-eac6-4377-8721-10456ead39a2)

I downloaded https://github.com/shashisadasivan/SSD365VSAddIn/tree/master/OutputDlls manually and I copied its content to Addin folder (having additional envdte80.dll and Microsoft.VisualStudio.Interop.dll and 'Unblock' property checked), Addins appears:
![image](https://github.com/shashisadasivan/SSD365VSAddIn/assets/17122845/ffd0f8f0-f0ec-402a-8ea4-14b385e17463)
 but I am getting error when any label functionality executed
![image](https://github.com/shashisadasivan/SSD365VSAddIn/assets/17122845/72484122-5123-4403-bae1-8f31bf39fef3)

I guess this is caused by different versions of linked libraries. Do you have any hints how to proceed please?
